cmd/swarmctl/network/common.go

```diff
@@ -14,31 +14,66 @@ import (
 func GetNetwork(ctx context.Context, c api.ControlClient, input string) (*api.Network, error) {
 	// GetService to match via full ID.
 	rg, err := c.GetNetwork(ctx, &api.GetNetworkRequest{NetworkID: input})
+
 	if err != nil {
-		// If any error (including NotFound), ListServices to match via ID prefix and full name.
-		rl, err := c.ListNetworks(ctx,
-			&api.ListNetworksRequest{
-				Filters: &api.ListNetworksRequest_Filters{
-					Names: []string{input},
-				},
-			},
-		)
+		rl, err := getNetworkByName(ctx, c, input)
 		if err != nil {
-			return nil, err
+			rl, e1 := getNetworkByPrefixedID(ctx, c, input)
+			if e1 == nil {
+				return rl, nil
+			}
```

Comment:
If `e1` is not nil, we will make it to the bottom of this function and return `nil, nil`.

I think this should be

```go
if e1 != nil {
        return nil, e1
}
```